docs/sdk/v3/guides/02-quoting.md

```diff
@@ -1,6 +1,113 @@
+
+```js reference title="Computing the Pool's address"
+https://github.com/Uniswap/examples/blob/22ae27bafdbff895ee2168584154626ef4af4d30/v3-sdk/quoting/src/example/Example.tsx#L13-L18
+```
+Since each *Uniswap V3 Pool* is uniquely identified by 3 characteristics (token in, token out, fee), we use those
+in combination with the address of the *PoolFactory* contract to compute the address of the **USDC - ETH** Pool.
+These parameters have already been defined in our configuration file:
+
+```ts reference title="Configuration Parameters"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/config.ts#L34-L39
+```
+
+
+### Setting up a reference to the Pool contract
+
+Now that we have the deployment address of the **USDC - ETH** Pool, we can construct an instance of an **ethers** `Contract` to interact with it:
+
+```js reference title="Setting up a reference to the Pool contract"
+https://github.com/Uniswap/examples/blob/e1fbf8612e2e7d0b86a25637cc75881ff809ca2e/v3-sdk/quoting/src/example/Example.tsx#L27-L31
+```
+
+To construct the *Contract* we just need to provide the address of the contract, its ABI and the provider that will carry out the RPC call for us.
+We can get access to the contract's ABI through the [@uniswap/v3-core](https://www.npmjs.com/package/@uniswap/v3-core) package:
+
+```js reference title="Uniswap V3 Pool smart contract ABI"
+https://github.com/Uniswap/examples/blob/22ae27bafdbff895ee2168584154626ef4af4d30/v3-sdk/quoting/src/example/Example.tsx#L7
+```
+
+
+### Getting Pool metadata from the Pool smart contact
+
+Having constructed our reference to the contract, we can now access its methods through our provider.
+We use a batch `Promise` call. This approach queries state data concurrently, rather than sequentially, to avoid out of sync data that may be returned if sequential queries are executed over the span of two blocks:
+
+
+```js reference title="Getting Pool metadata from the Pool smart contact"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/example/Example.tsx#L22
+```
+
+The return values of these methods will become inputs to the quote fetching function.
+
+### Setting up a reference to the Quoter contract
+
+Like we did for the Pool contract, we need to construct an instance of an **ethers** `Contract` for our Quoter contract in order to interact with it:
+```js reference title="Setting up a reference to the Quoter contract"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/example/Example.tsx#L32
+```
+
+We can get access to the contract's ABI through the [@uniswap/v3-periphery](https://www.npmjs.com/package/@uniswap/v3-periphery) package:
+
+```js reference title="Uniswap V3 Quoter smart contract ABI"
+https://github.com/Uniswap/examples/blob/22ae27bafdbff895ee2168584154626ef4af4d30/v3-sdk/quoting/src/example/Example.tsx#L6
+```
```

Comment:
Do we need this code ref? It's in the code example so anyone can look there, otherwise it's just a very long line that doesn't offer much

Reply:
Got it, good with that!